src/test/java/bio/terra/service/load/LoadDaoUnitTest.java

```diff
@@ -188,78 +208,60 @@ public void loadFilesTest() throws Exception {
   }
 
   @Test
-  public void serialLockTest() throws Exception {
+  void serialLockTest() {
     final String loadTag = LoadTagsUsedByTest.LOADTAG_SERIAL.getTag();
     final String flightX = FlightIdsUsedByTest.FLIGHT_X.getId();
     final String flightY = FlightIdsUsedByTest.FLIGHT_Y.getId();
+    final String flightZ = FlightIdsUsedByTest.FLIGHT_Z.getId();
+    final UUID datasetIdA = datasets.get(0).getId();
+    final UUID datasetIdB = datasets.get(1).getId();
 
-    boolean xlocks = tryLockLoad(loadTag, flightX);
-    assertTrue("x gets lock", xlocks);
-
-    xlocks = tryLockLoad(loadTag, flightX);
-    assertTrue("x gets lock again", xlocks);
-
-    boolean ylocks = tryLockLoad(loadTag, flightY);
-    assertFalse("y does not get lock", ylocks);
-
-    loadDao.unlockLoad(loadTag, flightX);
-
-    ylocks = tryLockLoad(loadTag, flightY);
-    assertTrue("y gets lock", ylocks);
+    assertThat(
+        "[Flight=X,dataset=A] gets lock",
+        loadDao.lockLoad(loadTag, flightX, datasetIdA),
+        loadLockedBy(flightX, datasetIdA));
```

Comment:
Is a custom matcher needed here? I think this is the same as
```suggestion
    assertThat(
        "[Flight=X,dataset=A] gets lock",
        loadDao.lockLoad(loadTag, flightX, datasetIdA)
            .lockedBy(flightX, datasetIdA));
```

Reply:
This would be an equivalent assertion, but I don't like asserting a truth-y condition on a complex object if I can help it… if the assertion fails, we're lacking in helpful debugging info in the error message rather than seeing the full Load object failing the condition.

But, a custom matcher does add complexity, so I don't mind going to a truth-y assertion if that's what reviewers prefer.

Reply:
Update: refactoring meant that this custom matcher was no longer valuable (let alone necessary), I removed it.